File: test/controllers/LoginControllerSuite.java
package controllers;

import play.mvc.Http.RequestBuilder;

import org.junit.Test;
import play.mvc.Result;
import play.test.Helpers;
import utils.WiFreeSuite;

import static org.junit.Assert.assertEquals;
import static play.test.Helpers.*;

public class LoginControllerSuite extends WiFreeSuite {

	@Test
	public void testLogin() {
		final RequestBuilder loginRequest = Helpers.fakeRequest().method(GET).uri("/login");
		final Result result = route(app, loginRequest);
		assertEquals(OK, result.status());
	}

	@Test
	public void testIndex() {
		final RequestBuilder loginRequest = Helpers.fakeRequest().method(GET).uri("/old-index");
		final Result result = route(app, loginRequest);
		assertEquals(OK, result.status());
	}
}

File: test/controllers/LoginControllerSuite.scala
package controllers

import org.junit.Assert.{assertEquals, assertTrue}
import org.junit.Test
import play.mvc.Http.Status.OK
import play.test.Helpers
import play.test.Helpers.{GET, contentAsString, route}
import utils.WiFreeSuite

import scala.collection.JavaConverters._

class LoginControllerSuite extends WiFreeSuite {
  @Test def loginWithoutErrors(): Unit = {
    val loginRequest =
      Helpers.fakeRequest
        .method(GET)
        .uri("/login")

    val authResult = route(WiFreeSuite.app, loginRequest)
    assertEquals(OK, authResult.status)
    val authResultStr = contentAsString(authResult)
    assertTrue(!authResultStr.contains("CredentialsException"))
  }

  @Test def loginWithErrors(): Unit = {
    val loginRequest =
      Helpers.fakeRequest
        .method(GET)
        .uri("/login?username=bret%40famly.co&error=CredentialsException")

    val authResult = route(WiFreeSuite.app, loginRequest)
    val authResultStr = contentAsString(authResult)
    assertEquals(OK, authResult.status)
    assertTrue(authResultStr.contains("CredentialsException"))
  }
}
